pkg/audit/manager.go

```diff
@@ -323,19 +324,31 @@ func (ucloop *updateConstraintLoop) update() {
 				return true, nil
 			default:
 				failure := false
-				// if constraint is in updatedLists, then update its violations
+				ctx := context.Background()
+				name := item.GetName()
+				namespace := item.GetNamespace()
+				namespacedName := types.NamespacedName{
+					Name:      name,
+					Namespace: namespace,
+				}
+				// get the latest constraint
+				err := ucloop.client.Get(ctx, namespacedName, &item)
```

Comment:
We should deepcopy item to avoid unexpected side effects in the future. Maybe the DeepCopyInto() method?

Reply:
done